sql/dd/dd_table.cc

```diff
@@ -2315,14 +2316,15 @@ static std::unique_ptr<dd::Table> create_dd_system_table(
     return nullptr;
 
   /*
-    During --initialize, and for inert tables, get the SE private data
-    from the SE, and store it in the dd_properties table at a later stage.
+    During --initialize, DDSE change and for inert tables, get the SE private
+    data from the SE, and store it in the dd_properties table at a later stage.
     Otherwise, get the SE private data from the 'dd_properties' table.
   */
   const System_tables::Types *table_type =
       System_tables::instance()->find_type(system_schema.name(), table_name);
   if (opt_initialize ||
-      (table_type != nullptr && *table_type == System_tables::Types::INERT)) {
+      (table_type != nullptr && *table_type == System_tables::Types::INERT) ||
+      dd::bootstrap::DD_bootstrap_ctx::instance().is_dd_engine_change()) {
```

Comment:
We are already in the `dd::` namespace here:
```suggestion
      bootstrap::DD_bootstrap_ctx::instance().is_dd_engine_change()) {
```